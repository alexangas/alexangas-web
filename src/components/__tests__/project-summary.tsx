import React from "react"
import renderer from "react-test-renderer"

import ProjectSummary from "../project-summary"

describe(`ProjectSummary`, () => {
  it(`renders project summary with description`, () => {
    const testRenderer = renderer.create(
      <ProjectSummary
        title="Project Title"
        description="The description of the project."
        tags={[`tagA`, `tagB`, `tagC`]}
      />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`renders project summary with excerpt`, () => {
    const testRenderer = renderer.create(
      <ProjectSummary
        title="Project Title"
        excerpt="The excerpt of the project."
        tags={[`tagA`, `tagB`, `tagC`]}
      />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`renders project summary with link`, () => {
    const testRenderer = renderer.create(
      <ProjectSummary
        title="Project Title"
        description="The description of the project."
        tags={[`tagA`, `tagB`, `tagC`]}
        link="website.alexangas.com"
      />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
