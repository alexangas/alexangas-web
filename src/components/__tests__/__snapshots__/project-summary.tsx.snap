// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSummary renders project summary with description 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p
        className="title is-4 is-marginless"
      >
        Project Title
      </p>
      
      <p
        className="is-marginless mb-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "The description of the project.",
          }
        }
      />
      <div
        className="tags is-inline-block-tablet"
      >
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagA
        </span>
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagB
        </span>
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagC
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectSummary renders project summary with excerpt 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p
        className="title is-4 is-marginless"
      >
        Project Title
      </p>
      
      <p
        className="is-marginless mb-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "The excerpt of the project.",
          }
        }
      />
      <div
        className="tags is-inline-block-tablet"
      >
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagA
        </span>
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagB
        </span>
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagC
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectSummary renders project summary with link 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p
        className="title is-4 is-marginless"
      >
        Project Title
      </p>
      <div
        className="my-2"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
          />
        </svg>
        <a
          className="ml-1"
          href="https://website.alexangas.com"
          rel="noreferrer noopener"
          target="_blank"
        >
          website.alexangas.com
        </a>
      </div>
      <p
        className="is-marginless mb-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "The description of the project.",
          }
        }
      />
      <div
        className="tags is-inline-block-tablet"
      >
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagA
        </span>
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagB
        </span>
        <span
          className="tag is-paddingless"
        >
          <span
            className="icon mr-0"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
          tagC
        </span>
      </div>
    </div>
  </div>
</div>
`;
